### PR TITLE
Buttons

### PR DESCRIPTION
**Pull request by rlisahuang**
- Reset button now placed to the right of the copy button
- To fix the positioning of the Github Discussion button for readonly blocks (z3-python), replaced the native code blocks with our own custom editor (readonly mode) for those blocks

### CODE REVIEW

**Comment by rlisahuang** [comment]
Only changes from https://github.com/microsoft/z3guide/pull/95/commits/cdaa09fcbbeb9fab3f391c96631e283dce6252f0 are relevant. Forgot to create a new branch from the old `custom-editor` branch.

**Comment by rlisahuang** [comment]
Editable blocks:
Screen Shot 2022-08-09 at 3 57 09 PM

Read-only blocks:
Screen Shot 2022-08-09 at 4 25 09 PM

**Comment by rlisahuang** [comment]
TODO:

After "Reset" is pressed, show an "Undo" button for 3 seconds, and make the editor read-only during that period

**Comment by rlisahuang** [comment]
Fixed #93 and #94 .

**Comment by rlisahuang** [comment]
New features:
- indication of runnable vs. read-only blocks
- language runtime name and info (for reporting problems easier if any, as is suggested in #65 )

Screen Shot 2022-08-12 at 3 07 24 PM

**Comment by rlisahuang** [comment]
Not sure if it is the best design to put everything inside the Run button. Happy to move it elsewhere.

**Comment by rlisahuang** [comment]
Feature: undo button (live for 3 seconds) after Reset is clicked
- on click -> revert changes back to before reset
- freeze the editor (read-only mode) during the reset/undo period
Screen Shot 2022-08-12 at 3 45 23 PM

**Comment by pelikhan** [comment]
Put version in footer?
________________________________

**Comment by rlisahuang** [comment]
The idea is to put the z3 version information somewhere around/close to the editor. People could run into problems that could be a bug of z3-solver, so it is better to have the version info handy. I don't know if it is obvious enough if it is placed in the footer.

Oh... maybe it could be part of the embedded/prefilled info when clicking on the "discussion" button?

**Comment by pelikhan** [comment]
Version number can typically be found in an about dialog.
________________________________

**Comment by pelikhan** [comment]
> Oh... maybe it could be part of the embedded/prefilled info when clicking on the "discussion" button?

Yes definitely
________________________________

**Comment by pelikhan** [comment]
image

or just don't show anything?

File: website/src/components/TutorialComponents/CodeBlock.tsx [hunk 1]
```diff
@@ -33,6 +35,46 @@ export function GithubDiscussionBtn({ repo }) {
...
+    return (
+        <button
+            type="button"
+            aria-label="Undo the reset"
+            title="Undo the reset"
+            className={clsx(
+                'clean-btn',
+                codeBlockContentStyles.codeButton,
+            )}
+            style={{borderColor: "#374BBB"}}
```
Comment (pelikhan) [inline website/src/components/TutorialComponents/CodeBlock.tsx:69]: works in dark mode?

File: website/src/components/TutorialComponents/CodeBlock.tsx [hunk 2]
```diff
@@ -61,7 +107,7 @@ function CodeEditor(props: {
     }, []);
 
     useEditable(editorRef, onEditableChange, {
-        disabled: props.disabled,
+        disabled: disabled,
```
Comment (pelikhan) [inline website/src/components/TutorialComponents/CodeBlock.tsx:110]: nit:

    disabled: disabled => disabled

File: website/src/components/TutorialComponents/index.tsx [hunk 3]
```diff
@@ -37,21 +37,32 @@ interface CodeBlockProps {
...
+    throw new Error('buggy code; read-only blocks should not have been attached with language executables');
+  }
+
+  const langInfo = version && tool ? ` (${tool}, ver. ${version})` : '';
+
   return (
-    <button className="button button--primary" onClick={props.onClick}>
-      Run
+    <button disabled={disabled} className="button button--primary" onClick={onClick}>
+      {props.disabled ? "This block is read-only" : `Run${langInfo}`}
```
Comment (pelikhan) [inline website/src/components/TutorialComponents/index.tsx:55]: IMO, don't show a button if run is not supported. keep it simple.

**Review by pelikhan — APPROVED** [review]

**Comment by pelikhan** [comment]
Don't bundle many features in a single PR; use multiple smaller PRs with a single feature.

File: website/src/components/TutorialComponents/CodeBlock.tsx (hunk 2 above)
Comment (rlisahuang) [reply website/src/components/TutorialComponents/CodeBlock.tsx:110]: What's the problem with using the boolean directly?

Comment (pelikhan) [reply website/src/components/TutorialComponents/CodeBlock.tsx:110]: it's just a TypeScript shorthand, where you can put the variable and TypeScript will use the variable name as the field name.

    { disabled: disabled }

to
    {disabled}

Comment (pelikhan) [reply website/src/components/TutorialComponents/CodeBlock.tsx:110]: nit = nit picking

**Comment by rlisahuang** [comment]
Ready for review and merge.

**Pull request merged**